lib/engine/step/g_18_zoo/track.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../track'
+
+module Engine
+  module Step
+    module G18ZOO
+      class Track < Track
+        def lay_tile(action, _extra_cost: 0, _entity: nil, _spender: nil)
+          hex = action.hex
+          tile = action.tile
+
+          if %w[O M MM].include?(hex.location_name) &&
+            tile.color == :yellow &&
+            !tile.name.end_with?('(water)') && !tile.name.end_with?('(mountain)')
+            tile = case tile.name + '_' + hex.location_name
+                   when '7_O'
+                     Engine::Game::G18ZOO::TILE_W7.dup
```

Comment:
what is this? why are you doing this? are you trying to make infinite tiles? you can just set that in the config if you are

Reply:
the tile 7,8,9 are changed into a new kind of tile (with town) when you put on the water terrain; so I don't have them as specific tile, but I'll change them on fly with a new one